plugins/optimization-detective/docs/hooks.md

```diff
@@ -198,16 +198,30 @@ Filters whether the current response can be optimized. By default, detection and
 
 1. It’s not a search template (`is_search()`).
 2. It’s not a post embed template (`is_embed()`).
-3. It’s not the Customizer preview (`is_customize_preview()`)
-4. It’s not the response to a `POST` request.
-5. There is at least one queried post on the page. This is used to facilitate the purging of page caches after a new URL Metric is stored.
+3. It’s not a preview (`is_preview()`).
+4. It’s not the Customizer preview (`is_customize_preview()`);
+5. It’s not the response to a `POST` request.
+6. There is at least one queried post on the page. This is used to facilitate the purging of page caches after a new URL Metric is stored.
 
-To force every response to be optimized regardless of the conditions above, you can do:
+The filter now receives an additional `$disabled_flags` parameter that contains information about which specific conditions are preventing optimization. This allows for more granular control.
+
+For example, to enable optimization specifically for search pages:
 
 ```php
-add_filter( 'od_can_optimize_response', '__return_true' );
+add_filter( 'od_can_optimize_response', function( $can_optimize, $disabled_flags ) {
+    if ( ! $can_optimize && isset( $disabled_flags['is_search'] ) && $disabled_flags['is_search'] ) {
```

Comment:
Now that the keys are defined for the `$disabled_flags` array, there is no need for the `isset()` check, right?

```suggestion
    if ( ! $can_optimize && $disabled_flags['is_search'] ) {
```